lib/base/log.c

```diff
@@ -199,38 +199,44 @@ open_syslog(heim_context context,
 }
 
 struct file_data {
-    const char *filename;
+    char *filename;
     const char *mode;
     struct timeval tv;
     FILE *fd;
     int disp;
 #define FILEDISP_KEEPOPEN       0x1
 #define FILEDISP_REOPEN         0x2
-#define FILEDISP_IFEXISTS       0x3
-    int freefilename;
+#define FILEDISP_IFEXISTS       0x4
 };
 
-static void HEIM_CALLCONV
+#ifndef O_CLOEXEC
+#define O_CLOEXEC 0
+#endif
+
+static void
 log_file(heim_context context, const char *timestr, const char *msg, void *data)
 {
     struct timeval tv;
     struct file_data *f = data;
+    FILE *logf = f->fd;
     char *msgclean;
-    size_t i;
+    size_t i = 0;
     size_t j;
 
-    if (f->disp != FILEDISP_KEEPOPEN) {
-        char *filename;
-        int flags = -1;
+    if (logf == NULL || (f->disp & FILEDISP_REOPEN)) {
+        int flags = O_WRONLY|O_APPEND;
         int fd;
 
-        if (f->mode[0] == 'w' && f->mode[1] == 0)
-            flags = O_WRONLY|O_TRUNC;
-        if (f->mode[0] == 'a' && f->mode[1] == 0)
-            flags = O_WRONLY|O_APPEND;
-        assert(flags != -1);
+        if (f->mode[0] == 'e') {
```

Comment:
Bug: only the last letter can be `'e'`.  It's a Linux-ism.

Also, maybe we should just call `fopen()` and not `open()` then `fdopen()`.

Reply:
Maybe we should have an `rk_fopen()` to handle the `'e'` in the mode.  (`'e'` == `O_CLOEXEC`).

Reply:
We can also drop the `'e'` flag business.